### PR TITLE
Remove content migrated to sigstore docs

### LINKED ISSUE #444: Migrate docs to docs.sigstore.dev
We now have a nice docs site, so let's start moving content into there.

Seeking volunteers to help with this and also add more content.

### PR DESCRIPTION
**Pull request by endorama**
#### Summary

Remove all content migrated to the Sigstore documentation website in https://github.com/sigstore/sigstore-website/pull/76

#### Ticket Link

Fixes https://github.com/sigstore/rekor/issues/444

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Comment by dlorenc** [comment]
DCO got us again!

**Comment by cpanato** [comment]
@endorama thanks for this PR, can you please sign the DCO?

**Comment by endorama** [comment]
My bad, I always forgot that when committing from GitHub I have to manually add Signed Off lines in the commits. Rebase in progress!

File: README.md [hunk 1]
```diff
@@ -14,20 +14,15 @@ Rekor fulfils the signature transparency role of sigstore's software signing
 infrastructure. However, Rekor can be run on its own and is designed to be
 extensible to working with different manifest schemas and PKI tooling.
 
-## Public Instance
+[Official Documentation](http://docs.sigstore.dev/rekor/overview).
```
Comment (bobcallaway) [inline README.md]:
```suggestion
[Official Documentation](https://docs.sigstore.dev/rekor/overview).
```

File: README.md [hunk 2]
```diff
@@ -14,20 +14,15 @@ Rekor fulfils the signature transparency role of sigstore's software signing
...
 We **will take the log down** and reset it with zero notice.
 We will improve the stability and publish SLOs over time.
 
-This instance is maintained by the Rekor community.
-Interested in helping operate and maintain our production CA system and Transparency Logs?
-Please reach out via the [mailing list](https://groups.google.com/g/sigstore-dev).
-
-If you have production use-cases in mind, again - please reach out over email via the [mailing list](https://groups.google.com/g/sigstore-dev).
-We are interested in helping on board you!
+More details on the public instance can be found at [docs.sigstore.dev](http://docs.sigstore.dev/rekor/public-instance).
```
Comment (bobcallaway) [inline README.md]:
```suggestion
More details on the public instance can be found at [docs.sigstore.dev](https://docs.sigstore.dev/rekor/public-instance).
```

**Review by bobcallaway — COMMENTED** [review]
2 minor nits to s/http/https but otherwise LGTM

**Comment by endorama** [comment]
@bobcallaway thank you for catching them, I wrote the URL manually and forgot the `s` :sweat_smile:

**Comment by endorama** [comment]
DCO is my worst enemy :D

**Review by bobcallaway — APPROVED** [review]

**Pull request merged**